### DISCUSSION TITLE
is vetur going to support vue 3 with typescript?

### DISCUSSION THREAD

**Discussion by JenuelDev** [vuejs/vetur#3183] [Q&A] [answered]
is vetur going to support vue 3 with typescript? what is happening to the vetur, why is everybody saying to volar instead of vetur? will vetur going to support what volar can do?

**Comment by yoyo930021** [comment, marked as answer]
You can use vue3 with typescript in Vetur, but don't support `<script setup>` now.
Volar is good for type-safe Vue templates. you can get more features for typescript.

In the future maybe the two plugins will merge or collaborate more.

**Reply by JenuelDev** [reply]
any update on this? is vetur can support script setup now?

**Reply by nikkorejz** [reply]
Any updates?

**Reply by boutell** [reply]
Evan You has said that "Volar is the new recommendation for Vue 3 in all cases." So I guess that's it.
